pkg/config/config.go

```diff
@@ -0,0 +1,158 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	sriovtypes "github.com/intel/sriov-cni/pkg/types"
+	"github.com/intel/sriov-cni/pkg/utils"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	defaultCNIDir = "/var/lib/cni/sriov"
+	// MaxSharedVf defines maximum number of PFs a VF is being shared
+	MaxSharedVf = 2
+)
+
+// LoadConf parses and validates stdin netconf and returns NetConf object
+func LoadConf(bytes []byte) (*sriovtypes.NetConf, error) {
+	n := &sriovtypes.NetConf{}
+	if err := json.Unmarshal(bytes, n); err != nil {
+		return nil, fmt.Errorf("failed to load netconf: %v", err)
+	}
+
+	// DeviceID takes precedence; if we are given a VF pciaddr then work from there
+	if n.DeviceID != "" {
+		// Get rest of the VF information
+		vfInfo, err := getVfInfo(n.DeviceID)
+		if err != nil {
+			return nil, err
+		}
+		n.DeviceInfo = vfInfo
+		n.Master = vfInfo.Pfname
```

Comment:
if we fix the comment to setupVF function below,  this line can be removed.
n.Master is only useful in free mode, for deviceID mode, DeviceInfo.Pfname can be used.

Reply:
The n.Master is used in many places so we need this value but going forward we can definitely review the NetConf struct and try consolidate all info into one place.